subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,609 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.transform
+
+import groovy.transform.EqualsAndHashCode
+import org.gradle.api.internal.artifacts.transform.ExecuteScheduledTransformationStepBuildOperationType
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType
+import org.gradle.internal.taskgraph.CalculateTaskGraphBuildOperationType.PlannedNode
+import org.gradle.internal.taskgraph.NodeIdentity
+
+import java.util.function.Predicate
+
+class ArtifactTransformBuildOperationIntegrationTest extends AbstractIntegrationSpec implements ArtifactTransformTestFixture {
+
+    @EqualsAndHashCode
+    static class NodeId {
+        String nodeType
+        String nodeIdInType
+    }
+
+    static class NodeMatcher {
+        String phantomId
+        String nodeType
+        Predicate<NodeIdentity> identityPredicate
+        List<String> dependencyPhantomIds = []
+
+        def matchNode(plannedNode) {
+            plannedNode.nodeIdentity.nodeType.toString() == nodeType && identityPredicate.test(plannedNode.nodeIdentity)
+        }
+    }
+
+    static final Set<String> KNOWN_NODE_TYPES = NodeIdentity.NodeType.values().collect { it.name() } as Set<String>
+    static final String TASK = NodeIdentity.NodeType.TASK.name()
+    static final String ARTIFACT_TRANSFORM = NodeIdentity.NodeType.ARTIFACT_TRANSFORM.name()
+
+    def buildOperations = new BuildOperationsFixture(executer, testDirectoryProvider)
+
+    def setup() {
+        requireOwnGradleUserHomeDir()
+
+        buildFile << """
+            allprojects {
+                group = "colored"
+            }
+        """
+    }
+
+    def setupExternalDependency() {
+        def m1 = mavenRepo.module("test", "test", "4.2").publish()
+        m1.artifactFile.text = "test-test"
+
+        buildFile << """
+            allprojects {
+                group = "colored"
```

Comment:
Why do you change the group name here?

Reply:
It was a copy-paste artifact. Remove it from here.